src/controllers/AuthController.js

```diff
@@ -105,6 +105,146 @@ class AuthController {
       picture: user.photos[0].value,
     };
   }
+
+  /**
+   * signup only for organizations
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} res
+   * @memberof Auth
+   */
+  static async signup(req, res) {
+    const hashedPassword = encrypt.hashPassword(req.body.password);
+    const { companyName, username, email } = req.body;
+
+    const ret = field =>
+      res.status(400).json({ message: `${field} already exist` });
+
+    const existUsername = await User.findOne({
+      username,
+    });
+    const existEmail = await User.findOne({
+      email,
+    });
+
+    if (existUsername) {
+      return ret('Username');
+    }
+    if (existEmail) {
+      return ret('Email');
+    }
+
+    const user = await User.create({
+      username,
+      email,
+      password: hashedPassword,
+    });
+    const organization = await Organization.create({
+      name: companyName,
+      user: user.id,
+    });
+
+    const getUser = id =>
+      Organization.findById(id)
+        .populate('user')
+        .exec();
+
+    const result = await getUser(organization.id);
+
+    const token = encrypt.generateToken({ email, username });
+    if (process.env.NODE_ENV !== 'test') {
```

Comment:
@Karlmusingo Why this?

Reply:
To not try to send emails when running test

Reply:
@Karlmusingo Test should not be handled in your logic, but it should be mocked instead